### PR TITLE
Admin pages for Advice

### PR DESCRIPTION
**Pull request by XuanLi-Leong**
## Status: 

:rocket: Ready

## Description
Add & Edit/Delete pages for parent advice
+ Catch regarding active RO year 
+ Some random typo/copy pasta fixes

Related issues: #237 #122

### CODE REVIEW

File: frontend/src/pages/admin/AdminEditAdvicePage.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            {!this.state.success && (
+              <AdminAdviceSelect handleSelect={this.handleSelect} />
+            )}
+            {!this.state.success &&
+              (this.state.selectedAdvice !== null && (
+                <AdminAdviceForm
+                  type="Edit"
+                  handleSuccess={this.handleSuccess}
+                  advice={this.state.selectedAdvice}
+                  key={this.state.selectedAdvice.id}
```
Comment (XuanLi-Leong) [inline frontend/src/pages/admin/AdminEditAdvicePage.js]: @josh-byster You might be interested in this, I was reading the React documentation about `getDerivedStateFromProps` and saw their suggestion of using a key. When the key changes, React constructs a new component so the constructor gets called again! By doing this, I didn't have to write an update function inside `AdminAdviceForm` to handle when the selected advice changed.

File: frontend/src/utils/api.js [hunk 2]
```diff
@@ -28,7 +28,10 @@ function login(formData) {
 function search(searchString) {
   const query = `search_string=${searchString}`;
   return fetch(`${API_URL}/books?${query}`, {
-    method: 'GET'
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json'
```
Comment (XuanLi-Leong) [inline frontend/src/utils/api.js]: Added this to all the API calls for consistency

File: frontend/src/utils/api.js [hunk 3]
```diff
@@ -87,13 +86,20 @@ function publishAllBooksInYear(readingOlympicsYear) {
   };
   // return fetch(`${API_URL}/setYearOrSomething`, {
   //   method: 'POST',
+  // headers: {
+  //   'Content-Type': 'application/json'
+  // },
+  // credentials: 'include',
```
Comment (XuanLi-Leong) [inline frontend/src/utils/api.js]: Oh @Davidw1339 we need an endpoint to publish all books in a year..

File: frontend/src/pages/admin/AdminEditAdvicePage.js (hunk 1 above)
Comment (josh-byster) [reply frontend/src/pages/admin/AdminEditAdvicePage.js]: whattttttt that's so cool! thanks for sharing!!

File: frontend/src/utils/api.js (hunk 3 above)
Comment (Davidw1339) [reply frontend/src/utils/api.js]: oh yup that should be through route /publish_books which takes in a year and boolean to publish or unpublish. https://github.com/hack4impact-uiuc/philareads/blob/master/backend/api/views/book.py#L251

Json request should be along the lines of
```JSON
{
    "year": 2019,
    "published": true
}
```
to publish all books in 2019

File: backend/api/views/year.py [hunk 4]
```diff
@@ -18,8 +18,7 @@ def init_year():
 
 
 @year.route("/year", methods=["GET"])
-@admin_route
-def get_year(user_id):
+def get_year():
```
Comment (Davidw1339) [inline backend/api/views/year.py]: oh haha sorry about this 😥

**Pull request merged**